src/box/replication.cc

```diff
@@ -1045,9 +1060,8 @@ replicaset_connect(struct applier **appliers, int count,
 		/* Timeout or connection failure. */
 		if (state.connected < replicaset_connect_quorum(count) &&
 		    connect_quorum) {
-			diag_set(ClientError, ER_CFG, "replication",
-				 "failed to connect to one or more replicas");
-			goto error;
+			tnt_raise(ClientError, ER_CFG, "replication",
```

Comment:
Can't catch why you don't trigger_clear(&triggers[i].base);

Reply:
It isn't necessary because we call `applier_delete` in the guard destructor, which destroys all triggers.